### PR TITLE
[Impeller] fold memory check into allocator_vk

### PR DESCRIPTION
**Pull request by jonahwilliams**
Various cleanups to Vulkan allocator implementation:

1. Fixes https://github.com/flutter/flutter/issues/137454
2. Fold device transient cap check into allocator.
3. adds debug tracking for total memory usage in MB (a followup change needs to be made to driver to plumb it through)
4. Small cleanups to mock vulkan so an allocator can be created from it.
5. depth/stencil shouldn't be input attachments.

Part of https://github.com/flutter/flutter/issues/144617

### CODE REVIEW

File: impeller/renderer/backend/vulkan/allocator_vk.cc [hunk 1]
```diff
@@ -149,11 +228,25 @@ AllocatorVK::AllocatorVK(std::weak_ptr<Context> context,
     VALIDATION_LOG << "Could not create memory allocator";
     return;
   }
+
+  {
+    // Query texture support.
+    // TODO(jonahwilliams):
+    // https://github.com/flutter/flutter/issues/129784
+    physical_device.getMemoryProperties(&memory_properties_);
```
Comment (matanlurey) [inline impeller/renderer/backend/vulkan/allocator_vk.cc:236]: Wasn't this already done above on line 182? Or is the scope significant somehow?

File: impeller/renderer/backend/vulkan/allocator_vk.cc [hunk 2]
```diff
@@ -149,11 +228,25 @@ AllocatorVK::AllocatorVK(std::weak_ptr<Context> context,
...
+
+    for (auto i = 0u; i < memory_properties_.memoryTypeCount; i++) {
+      if (memory_properties_.memoryTypes[i].propertyFlags &
+          vk::MemoryPropertyFlagBits::eLazilyAllocated) {
+        supports_memoryless_textures_ = true;
```
Comment (matanlurey) [inline impeller/renderer/backend/vulkan/allocator_vk.cc:241]: `break` after since it doesn't seem relevant to iterate through once we find one?

**Review by matanlurey — APPROVED** [review]
It was a little tough to review 5 changes at once, especially as I'm not super familiar. If these changes aren't interconnected I'd recommend splitting them up somewhat, if they are then I suppose it's fine 😅

File: impeller/renderer/backend/vulkan/allocator_vk.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/allocator_vk.cc:236]: oh good call. Will fix

File: impeller/renderer/backend/vulkan/allocator_vk.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/allocator_vk.cc:241]: Done

File: impeller/renderer/backend/vulkan/allocator_vk.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/vulkan/allocator_vk.cc:236]: Done

**Comment by jonahwilliams** [comment]
revert reason: unexpected benchmark regression

https://flutter-flutter-perf.skia.org/e/?queries=device_type%3DPixel_7_Pro%26sub_result%3D90th_percentile_frame_rasterizer_time_millis%26sub_result%3D99th_percentile_frame_rasterizer_time_millis%26sub_result%3Daverage_frame_rasterizer_time_millis%26sub_result%3Dworst_frame_rasterizer_time_millis%26test%3Dnew_gallery_impeller__transition_perf&selected=commit%3D39690%26name%3D%252Carch%253Dintel%252Cbranch%253Dmaster%252Cconfig%253Ddefault%252Cdevice_type%253DPixel_7_Pro%252Cdevice_version%253Dnone%252Chost_type%253Dlinux%252Csub_result%253Dworst_frame_rasterizer_time_millis%252Ctest%253Dnew_gallery_impeller__transition_perf%252C

**Comment by jonahwilliams** [comment]
Reason for revert: unexpected benchmark regression

https://flutter-flutter-perf.skia.org/e/?queries=device_type%3DPixel_7_Pro%26sub_result%3D90th_percentile_frame_rasterizer_time_millis%26sub_result%3D99th_percentile_frame_rasterizer_time_millis%26sub_result%3Daverage_frame_rasterizer_time_millis%26sub_result%3Dworst_frame_rasterizer_time_millis%26test%3Dnew_gallery_impeller__transition_perf&selected=commit%3D39690%26name%3D%252Carch%253Dintel%252Cbranch%253Dmaster%252Cconfig%253Ddefault%252Cdevice_type%253DPixel_7_Pro%252Cdevice_version%253Dnone%252Chost_type%253Dlinux%252Csub_result%253Dworst_frame_rasterizer_time_millis%252Ctest%253Dnew_gallery_impeller__transition_perf%252C
